storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -96,57 +117,121 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		namespace   = ns.ID()
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
-		bopts       = s.opts.ResultOptions()
-		blopts      = bopts.DatabaseBlockOptions()
-		blockSize   = ns.Options().RetentionOptions().BlockSize()
-		encoderPool = bopts.DatabaseBlockOptions().EncoderPool()
-		errs        = 0
+		namespace    = ns.ID()
+		numShards    = len(shardsTimeRanges)
+		highestShard = s.findHighestShard(shardsTimeRanges)
+		numConc      = s.opts.EncodingConcurrency()
+		bopts        = s.opts.ResultOptions()
+		blopts       = bopts.DatabaseBlockOptions()
+		blockSize    = ns.Options().RetentionOptions().BlockSize()
+		encoderPool  = bopts.DatabaseBlockOptions().EncoderPool()
+		workerErrs   = make([]int, numConc)
 	)
-	for iter.Next() {
-		series, dp, unit, annotation := iter.Current()
 
-		// check if the series belongs to current namespace being bootstrapped
-		if !namespace.Equal(series.Namespace) {
-			continue
+	fmt.Println("Highest shard: ", highestShard)
```

Comment:
do you need to drop this println?

Reply:
...yes. Removed it. Maybe I should write a linter for that to...

Reply:
lol. I usually put a `TODO(prateek): ` when i leave something around and then it's obvious I have to clean it up.